### PR TITLE
USB: harness: Improve testability

### PR DESCRIPTION
**Pull request by finikorg**
Enable for `native_posix` `usb_device`. Switch on more tests, fix issues found in the tests.

Command to execute tests:
```
sanitycheck --device-testing --device-serial /dev/ttyACM0 -p <board> -t usb
```

### CODE REVIEW

**Review by loicpoulain — APPROVED** [review]

**Comment by carlescufi** [comment]
@niko @masz-nordic can you run this on nrf52840_pca10056?

**Review by aescolar — COMMENTED** [review]
I'm ok with the native_posix changes.

**Comment by masz-nordic** [comment]
nrf52840_pca10056 fails following tests:
- samples/subsys/usb/dfu/test - timeout
handler.log:
```
***** Booting Zephyr OS v1.14.0-rc1-1399-g0002598c309c *****

[00:00:00.000,000] <dbg> main.composite_pre_init: HID Device: dev 0x20003724

[00:00:00.000,000] <dbg> main.main: Starting application

[00:00:00.003,509] <dbg> main.status_cb: status 2 unhandled

[00:00:00.006,622] <dbg> main.status_cb: status 5 unhandled
```
- tests/subsys/usb/device/usb.device - timeout
handler.log is empty
- tests/subsys/usb/bos/usb.bos - timeout
handler.log is empty

**Comment by carlescufi** [comment]
@finikorg thanks for this!

There are 2 failures with nRF52840. One is expected (DFU timeout) but the other one is not:

```
$ scripts/sanitycheck --device-testing --device-serial /dev/ttyACM0 -p nrf52840_pca10056 -t usb
JOBS: 16
Building testcase defconfigs...
12 tests selected, 92405 tests discarded due to filters
total complete:    4/  12  33%  failed:    0

nrf52840_pca10056         samples/subsys/usb/dfu/test                        FAILED: timeout
        see: sanity-out/nrf52840_pca10056/samples/subsys/usb/dfu/test/handler.log

total complete:    9/  12  75%  failed:    1

nrf52840_pca10056         tests/subsys/usb/device/usb.device                 FAILED: timeout
        see: sanity-out/nrf52840_pca10056/tests/subsys/usb/device/usb.device/handler.log

total complete:   12/  12  100%  failed:    2
10 of 12 tests passed with 0 warnings in 223 seconds
```

**Comment by masz-nordic** [comment]
I have tried rerunning tests for a few times - they seem very inconsistent. I get from 2 up to 6 failures depending on rerun.

**Review by nashif — APPROVED** [review]

File: tests/subsys/usb/device/src/main.c [hunk 1]
```diff
@@ -158,7 +158,6 @@ static void test_usb_deconfig(void)
 /* Test USB Device Cotnroller API */
 static void test_usb_dc_api(void)
 {
-	/* Control endpoins are configured */
```
Comment (jfischer-phytec-iot) [inline tests/subsys/usb/device/src/main.c]: Something is wrong with this commit, it does not change any functionality in the code.

**Review by jfischer-no — CHANGES_REQUESTED** [review]

File: tests/subsys/usb/device/src/main.c (hunk 1 above)
Comment (finikorg) [reply tests/subsys/usb/device/src/main.c]: Fixed, somehow the patch was incorrectly rebased.

**Review by jfischer-no — APPROVED** [review]

**Pull request merged**